File: python/66_Plus_One.py
class Solution:  
    # @param digits, a list of integer digits  
    # @return a list of integer digits  
    def plusOne(self, digits):  
        len_s = len(digits)  
        carry = 1  
        for i in range(len_s - 1, -1, -1):  
            total = digits[i] + carry  
            digit = int(total % 10)  
            carry = int(total / 10)  
            digits[i] = digit  
        if 1 == carry:  
            digits.insert(0, 1)  
        return digits  

File: c++/66_Plus_One.cpp
/*************************************************************************
	> File Name: c++/66_Plus_One.cpp
	> Author: Pelhans 
	> Mail: me@pelhans.com
	> Created Time: 2018年04月02日 星期一 10时34分47秒
 ************************************************************************/

#include<iostream>
using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        add(digits, 1);
        return digits;
    }

private:
    void add( vector<int> &digits, int digit){
        int c = digit; //进位
        
        for (auto it = digits.rbegin(); it != digits.rend(); ++it){
            *it += c;
            c = *it / 10;
            *it %= 10;
        }
        if (c > 0) digits.insert(digits.begin(), 1);
    }
};
